Patch-Source: https://gitlab.alpinelinux.org/alpine/apk-tools/-/merge_requests/89
needed with https://github.com/bminor/musl/commit/25e6fee27f4a293728dd15b659170e7b9c7db9bc
--
From 2a9c0277c0d633d940a48726b214b89de2957d07 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Mon, 27 Dec 2021 13:57:13 -0600
Subject: [PATCH 2/2] use fstatat, not fstatat64

the stat64 family of functions were provided as transitional functions,
but when building on glibc with _GNU_SOURCE, or any other supported system,
the stat functions are equivalent to their stat64 counterparts
---
 src/database.c | 6 +++---
 src/io.c       | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/database.c b/src/database.c
index 7118c64..42a205e 100644
--- a/src/database.c
+++ b/src/database.c
@@ -1365,12 +1365,12 @@ static void handle_alarm(int sig)
 static char *find_mountpoint(int atfd, const char *rel_path)
 {
 	struct mntent *me;
-	struct stat64 st;
+	struct stat st;
 	FILE *f;
 	char *ret = NULL;
 	dev_t dev;
 
-	if (fstatat64(atfd, rel_path, &st, 0) != 0)
+	if (fstatat(atfd, rel_path, &st, 0) != 0)
 		return NULL;
 	dev = st.st_dev;
 
@@ -1380,7 +1380,7 @@ static char *find_mountpoint(int atfd, const char *rel_path)
 	while ((me = getmntent(f)) != NULL) {
 		if (strcmp(me->mnt_fsname, "rootfs") == 0)
 			continue;
-		if (fstatat64(atfd, me->mnt_dir, &st, 0) == 0 &&
+		if (fstatat(atfd, me->mnt_dir, &st, 0) == 0 &&
 		    st.st_dev == dev) {
 			ret = strdup(me->mnt_dir);
 			break;
diff --git a/src/io.c b/src/io.c
index 0a441bc..e9458cb 100644
--- a/src/io.c
+++ b/src/io.c
@@ -714,7 +714,7 @@ void apk_fileinfo_hash_xattr(struct apk_file_info *fi)
 int apk_fileinfo_get(int atfd, const char *filename, unsigned int flags,
 		     struct apk_file_info *fi, struct apk_atom_pool *atoms)
 {
-	struct stat64 st;
+	struct stat st;
 	unsigned int checksum = flags & 0xff;
 	unsigned int xattr_checksum = (flags >> 8) & 0xff;
 	int atflags = 0;
@@ -725,7 +725,7 @@ int apk_fileinfo_get(int atfd, const char *filename, unsigned int flags,
 	if (flags & APK_FI_NOFOLLOW)
 		atflags |= AT_SYMLINK_NOFOLLOW;
 
-	if (fstatat64(atfd, filename, &st, atflags) != 0)
+	if (fstatat(atfd, filename, &st, atflags) != 0)
 		return -errno;
 
 	*fi = (struct apk_file_info) {
